main.tf

```diff
@@ -0,0 +1,196 @@
+#
+# IAM resources
+#
+data "aws_iam_policy_document" "ecs_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "ecs_service_role" {
```

Comment:
I did consider this, but in the two roles contained within this module, the set of permissions they need are relatively fixed. Changes in the way these roles are used by ECS should occur transparently through updates in the referenced polices (`AmazonEC2ContainerServiceRole` and `AmazonEC2ContainerServiceAutoscaleRole `). If more than that is needed, I'd prefer to make changes to the module then and bump the version vs. building in an escape hatch via an `ecs_service_role` output.